packages/convex-helpers/server/pagination.ts

```diff
@@ -0,0 +1,301 @@
+import { Value, convexToJson } from "convex/values";
+import {
+  DocumentByName,
+  GenericDataModel,
+  GenericDatabaseReader,
+  IndexNames,
+  NamedTableInfo,
+  SchemaDefinition,
+  TableNamesInDataModel,
+} from "convex/server";
+
+export type IndexKey = Value[];
+
+export type PageRequest<
+  DataModel extends GenericDataModel,
+  T extends TableNamesInDataModel<DataModel>,
+> = {
+  /// Request a page of documents from this table.
```

Comment:
I know this is annoying, but `///` doesn't automatically generate docs on hover, at least for me. These need to be `/** Request a page of documents from this table. */` for single line, or `/** one line` ` * next line` ` */` style. VSCode should auto-generate some jsdoc syntax for functions. The nice thing is that then you get docs not only for functions but also for fields like this.

I've updated them this time - will push the commit